operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java

```diff
@@ -213,21 +221,40 @@ private static Secret buildSecretFrom(String name, String type, ManagedKafka man
         return secret;
     }
 
-    private static Secret kafkaTlsSecretFrom(ManagedKafka managedKafka, Secret current) {
-        return buildSecretFrom(kafkaTlsSecretName(managedKafka),
-                               "kubernetes.io/tls",
-                               managedKafka,
-                               current,
-                               Map.of("tls.crt", managedKafka.getSpec().getEndpoint().getTls().getCert(),
-                                      "tls.key", managedKafka.getSpec().getEndpoint().getTls().getKey()));
+    private static Secret kafkaTlsSecretFrom(ManagedKafka managedKafka, Secret current, Secret currentMasterSecret) {
+        if(currentMasterSecret != null){
+            return buildSecretFrom(kafkaTlsSecretName(managedKafka),
+                    "kubernetes.io/tls",
+                    managedKafka,
+                    current,
+                    Map.of("tls.crt", currentMasterSecret.getData().get("endpoint.tls.crt"),
+                            "tls.key", currentMasterSecret.getData().get("endpoint.tls.key")));
+        }
+        else {
+            return buildSecretFrom(kafkaTlsSecretName(managedKafka),
+                    "kubernetes.io/tls",
+                    managedKafka,
+                    current,
+                    Map.of("tls.crt", managedKafka.getSpec().getEndpoint().getTls().getCert(),
+                            "tls.key", managedKafka.getSpec().getEndpoint().getTls().getKey()));
+        }
     }
```

Comment:
Only a suggestion, but I think it's easier to read if only the parts that depend on the condition are in the if/then/else blocks.
```suggestion
        String cert;
        String key;
        
        if (currentMasterSecret != null) {
            cert = currentMasterSecret.getData().get("endpoint.tls.crt");
            key = currentMasterSecret.getData().get("endpoint.tls.key");
        } else {
            cert = managedKafka.getSpec().getEndpoint().getTls().getCert();
            key = managedKafka.getSpec().getEndpoint().getTls().getKey();
        }

        return buildSecretFrom(kafkaTlsSecretName(managedKafka),
                "kubernetes.io/tls",
                managedKafka,
                current,
                Map.of("tls.crt", cert,
                       "tls.key", key));
    }
```

Reply:
Sure! I could change the if/then/else blocks accordingly.